.config/s3.config.php

```diff
@@ -0,0 +1,20 @@
+<?php
+if (getenv('OBJECTSTORE_S3_HOST')) {
+  $CONFIG = array (
+    'objectstore' => array(
+      'class' => '\\OC\\Files\\ObjectStore\\S3',
+      'arguments' => array(
+        'bucket' => getenv('OBJECTSTORE_S3_BUCKET'),
+        'autocreate' => getenv('OBJECTSTORE_S3_AUTOCREATE') ?: true,
+        'key'    => getenv('OBJECTSTORE_S3_KEY'),
+        'secret' => getenv('OBJECTSTORE_S3_SECRET'),
+        'hostname' => getenv('OBJECTSTORE_S3_HOST'),
+        'port' => getenv('OBJECTSTORE_S3_PORT'),
+        'use_ssl' => getenv('OBJECTSTORE_S3_SSL') ?: true,
+        'region' => getenv('OBJECTSTORE_S3_REGION'),
+        // required for some non Amazon S3 implementations
+        'use_path_style' => getenv('OBJECTSTORE_S3_USEPATH_STYLE') ?: true,
```

Comment:
```suggestion
        'use_path_style' => getenv('S3_USE_PATH_STYLE') === 'true' ?: true,
```

I think I prefer the non amazon deployment ;)

Reply:
How does somebody get false here? I think this needs to be:

`'use_path_style' => getenv('OBJECTSTORE_S3_USEPATH_STYLE') === 'true' ?: false`

Reply:
I agree with using the explicit comparison as anything except `false`, `0`, and `null` will be true. If we want `OBJECTSTORE_S3_USEPATH_STYLE` to default to true, we can do this
```php
'use_path_style' => getenv('OBJECTSTORE_S3_USEPATH_STYLE') === 'false' ?: true,
```

Reply:
The server code defaults to "false" when nothing is specified. Here, you will require the container environment to say "false" to get it, else, it defaults to true....

https://github.com/nextcloud/server/blob/master/lib/private/Files/ObjectStore/S3ConnectionTrait.php#L115

If the environment variable is unset, do we want the logic to be flipped from how the base config processing works in the main nextcloud server repository?

Reply:
I think it makes the most sense to remain consistent. I don't feel strongly about this, so I'm fine either way as long as we agree.

Reply:
Here's the one I was able to make work to be consistent with the main project:

https://github.com/cuppett/nextcloud-openshift-s2i/blob/master/autoconfig/objectstorage.config.php

Here's a snippet I used to try and run that down:

https://gist.github.com/cuppett/f25928cf91c7f0d229e515b96efdf4e6